roofit/roofitcore/src/RooHistPdf.cxx

```diff
@@ -383,6 +403,30 @@ double RooHistPdf::analyticalIntegral(Int_t code, const char* rangeName) const
 }
 
 
+bool RooHistPdf::forceAnalyticalInt(RooArgSet const& pdfObsList, const RooAbsArg& dep)
+{
+   bool isOkayForAnalyticalInt = false;
+
+   for (RooAbsArg * obs : pdfObsList) {
+      if(obs->dependsOn(dep)) {
+         // If the observable doesn't depend linearly on the integration
+         // variable we will not do analytical integration.
+         auto lvalue = dynamic_cast<RooAbsRealLValue const*>(obs);
+         if(!(lvalue && lvalue->isJacobianOK(dep))) return false;
```

Comment:
I see that isJacobianOK returns by default `true`, i.e their derivative is constant. Can we make sure that some derived class implementing a more complex dependence (non constant Jacobian) will re-implement `isJacobianOK` ?

Reply:
Good question! We can't enforce it of course, but the case of inheriting from `RooAbsRealLValue` to implement a class that is at the same time settable and derived should be rare enough. If someone does that, they should really know what they are doing...